WordPress/Classes/Utility/Media/MediaImageExporter.swift

```diff
@@ -0,0 +1,257 @@
+import Foundation
+import MobileCoreServices
+
+/// MediaLibrary export handling of UIImages.
+///
+class MediaImageExporter: MediaExporter {
+
+    /// Default filename used when writing media images locally, which may be appended with "-1" or "-thumbnail".
+    ///
+    let defaultImageFilename = "image"
+
+    var maximumImageSize: CGFloat?
+    var stripsGeoLocationIfNeeded = false
+    var mediaDirectoryType: MediaLibrary.MediaDirectoryType = .uploads
+
+    public enum ExportError: MediaExportError {
+        case imageJPEGDataRepresentationFailed
+        case imageSourceCreationWithDataFailed
+        case imageSourceCreationWithURLFailed
+        case imageSourceIsAnUnknownType
+        case imageSourceExpectedJPEGImageType
+        case imageSourceDestinationWithURLFailed
+        case imageSourceThumbnailGenerationFailed
+        case imageSourceDestinationWriteFailed
+        var description: String {
+            switch self {
+            case .imageJPEGDataRepresentationFailed,
```

Comment:
Are you checking every case here to make sure we don't miss any that may be added in the future?